### ISSUE TITLE
BUG - sev-2- [iOS for sure, probably All] - Improve error handling for maintenance window call

### ISSUE THREAD

**Issue by TKDickson** [department-of-veterans-affairs/va-mobile-app#9170]
## What happened?

I was attempting to log into the mobile app in iOS in staging. When I got an error response back from the maintenance windows call, I was stuck indefinitely - we neither attempted to recall the MW API, nor was I allowed to proceed with login.

Just...... stayed on this screen for several minutes (I wanted to see if ~5 minutes later we'd try calling it again), before forceclosing the app.
<img width="371" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMXdHQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--aa161a370ca5891c0a428ac0bbd0960f2cd65327/image.png" alt="image.png" />

For the mobile/v0/maintenance_windows call, I got this response:
```
HTTP/1.1 504 Gateway Time-out
Date: Wed, 24 Jul 2024 14:25:42 GMT
Content-Type: text/html
Content-Length: 132
Connection: keep-alive

<html>
<head><title>504 Gateway Time-out</title></head>
<body>
<center><h1>504 Gateway Time-out</h1></center>
</body>
</html>

```

## Specs:

- Device:
- OS:
- User Account:
- Accessibility State:

## Desired behavior

Should gracefully handle errors. I'd recommend attempting to call the API again (maybe once?) and if we don't get a response back, either allow login but show some sort of error message within the app ("Hey things may be broken") or just assume that there are no maintenance windows in effect and allow API calls.

## Acceptance Criteria

- When an error response is returned from the maintenance window during login, retry the maintenance window call:
   - if the call fails again, then replace the home screen with the generic error screen 

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: High 
- Frequency: Low 

- 2 - High

## Screen shot(s) and additional information

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  

</details>

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [ ] Labels added (front-end, back-end, global, Health, relevant feature, accessibility, etc)
- [ ] Estimate of 1 added (for front-end tickets)
- [ ] Added either to the relevant feature epic (if found during new feature implementation), or the relevant team's bug epic ([Global](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8302), [Health & Benefits](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8303), [API](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8304), [QART](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8305))

**Comment by TKDickson** [comment]
OK I continued investigating, and I'm removing mentions of other APIs from this ticket. v2/user errors are handled just fine. v0/authorized-services are... better than maintenance windows, but with infinite reattempts on the profile screen - that's ticket

**Comment by dumathane** [comment]
Quick look at this and i'm missing context. dispatchSetDowntime unconditionally sets up state.downtimeWindowsFetched as true which should allow you to login. I need a better way to reproduce this on command to make sure I'm fully understanding this one. @TKDickson can you circle up with me about this? Set us a meeting next week?

**Comment by TKDickson** [comment]
Testing plan:
- [ ] Using a breakpoint to mimic the failure in ticket, a user can log in
- [ ] User is able to call all APIs (we will act like there are no maintenance windows, this was an AC change discussed with the team this week)
- [ ] Using a breakpoint, set maintenance windows and log back in as user.
- [ ] UI and API calls change appropriately when in downtime
- [ ] Using a breakpoint, set a more common API error (not the communication layer one in the ticket) for the maintenance window, and log back in as user.
- [ ] User is able to call all APIs (we will act like there are no maintenance windows, this was an AC change discussed with the team this week)
- [ ] Make sure that the fix for #8432 (which is the code that caused this bug) is still in place

**Comment by TKDickson** [comment]
Doesn't require further testing in develop, closing
